source/jormungandr/jormungandr/scenarios/journey_filter.py

```diff
@@ -647,6 +647,20 @@ def apply_final_journey_filters(response_list, instance, request):
     _filter_too_much_connections(journeys, instance, request)
 
 
+def replace_bss_tag(journeys):
+    """
+    replace the gss tag by walking, if there's no bss section in the journey
+    """
+    for j in journeys:
+        if not j.tags or "bss" not in j.tags:
+            continue
+        has_bss = any((s.type == response_pb2.BSS_RENT for s in j.sections))
+        if has_bss:
+            continue
+        j.tags.remove("bss")
+        j.tags.append("walking")
```

Comment:
Can it happen that there is no walking at all (e.g. the starting point is a stop point) and thus the "walking" tag should not be appended ?

Reply:
the goal is to replace the bss tag by walking ONLY when `bss` tag is existent in `journeys.tags`. In fact, with the presence of `bss` tag, we're sure that there must be a fallback return by the `bss` connector but we're not sure if it's pure walking or bss...

Reply:
Ok, I guess the case I am thinking about is the following : you give a stop_point as the starting point and ask for a `bss` fallback.
In this case, I expect the `bss` fallback to be empty (maybe I am wrong here), but it should be tagged as `bss`. 
So here, you remove the `bss` tag and add the `walking` tag. But the fallback has no walking at all.

Reply:
this case is protected by[ this line](https://github.com/CanalTP/navitia/pull/3478/files/14f6a6e217d8ab86ee93ddf80bbb623030047a99#diff-51842a7f082fe7a1d3dfc36f56a6370e922036194f379d51f0c92c497dc55a69R655)